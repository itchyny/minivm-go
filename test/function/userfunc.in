func bar(x)
  if x > 10
    return 30
  elseif x <= 0
    return -1
  end
end

print bar(-100)
print bar(5)
print bar(100)

func foo(argx, argy)
  print argx
  print argy
  print argx * argy
  hoge = 3 * argx + 4 * argy
  return 5 * hoge + (argy - argx)
end

print foo(foo(10, 20), 30)
print foo(foo(100, 200) - 300, 400)
