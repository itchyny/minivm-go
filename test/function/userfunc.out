-1
0
30
10
20
200
560
30
16800
8470
100
200
20000
5300
400
2120000
82600
